pyomo/contrib/mindtpy/util.py

```diff
@@ -641,3 +642,49 @@ def update_gap(solve_data):
     solve_data.abs_gap = solve_data.UB - solve_data.LB
     solve_data.rel_gap = (solve_data.UB - solve_data.LB)/(abs(
         solve_data.UB if solve_data.objective_sense == minimize else solve_data.LB) + 1E-10)
+
+
+def update_dual_bound(solve_data, bound_value):
+    '''
+    Call after solve relaxed problem, including relaxed NLP and MIP master problem.
+    Use the optimal primal bound of the relaxed problem to update the dual bound.
+    '''
+    if math.isnan(bound_value):
+        return
+    if solve_data.objective_sense == minimize:
+        solve_data.LB = max(bound_value, solve_data.LB)
+        solve_data.bound_improved = solve_data.LB > solve_data.LB_progress[-1]
+        solve_data.LB_progress.append(solve_data.LB)
+    else:
+        solve_data.UB = min(bound_value, solve_data.UB)
+        solve_data.bound_improved = solve_data.UB < solve_data.UB_progress[-1]
+        solve_data.UB_progress.append(solve_data.UB)
+    if solve_data.bound_improved:
+        update_gap(solve_data)
+
+
+def update_dual_bound_use_bound(solve_data, results):
```

Comment:
the name of this function is not very informative of what it does. How about `uptade_suboptimal_dual_bound'?